Blog: Why attacking your competitors online is dumb (#16704)
PostHog/posthog.com: contents/blog/why-attacking-competitors-is-dumb.md

---

> title: Why attacking your competitors online is dumb

Reviewer: ```suggestion
title: Attacking your competitors online is dumb
```

Author: I'm going to respectfully decline just this one, as I want the title to be quite opinionated rather than just explanatory

> title: Why attacking your competitors online is dumb
> date: 2026-05-02

Reviewer 2: ```suggestion
title: Why attacking your competitors online is dumb
date: 2026-05-07
```

author:
  - charles-cook
showTitle: true
rootPage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: >-
  https://res.cloudinary.com/dmukukwp6/image/upload/brand_startup_1aa2b151d6.png
featuredImageType: full
category: Startups
tags:
  - Guides
  - Marketing
  - Growth
  - Culture
---

> Let's say a competitor copies your feature. Let's say they lift your positioning, write a misleading comparison post, or just take a cheap shot at you on X. And let's say you're 'right' – they actually did the thing, and it's actually annoying.

Reviewer: ```suggestion
Let's say a competitor copies your feature. They lift your positioning, write a misleading comparison post, or just take a cheap shot at you on X. And let's say you're 'right' – they actually did the thing, and it's actually annoying.
```

> Let's say a competitor copies your feature. Let's say they lift your positioning, write a misleading comparison post, or just take a cheap shot at you on X. And let's say you're 'right' – they actually did the thing, and it's actually annoying.

Reviewer 2: ```suggestion
Let's say a competitor copies your feature. Let's say they lift your positioning, write a misleading comparison post, or just takes a cheap shot at you on X. And let's say you're right – they actually _did_ do whatever you believe they did, and you want to tell the internet about it.
```

> Still don't do it.

Reviewer 2: ```suggestion
Still, don't do it.
```

> This isn't a moral argument – I'm not going to lecture you about keeping it classy. I just think public competitor attacks never work, and there are much better places to put the energy.

Reviewer: ```suggestion
This isn't a moral argument about keeping it classy. I just think public competitor attacks never work, and there are much better places to put the energy.
```

> This isn't a moral argument – I'm not going to lecture you about keeping it classy. I just think public competitor attacks never work, and there are much better places to put the energy.

Reviewer 2: ```suggestion
This isn't a moral argument. I'm not going to lecture you about keeping it classy. It's just that public competitor attacks never work, and there are much better places to put that energy.
```

**TL;DR:**
- Even when you're right, it usually backfires
- You're amplifying a competitor who may have far fewer users than you
- Most of your market didn't see the thing you're upset about
- There are more effective ways to compete

> ---

Reviewer: ```suggestion
```

not needed. it's a short article anyway

## You're only talking to your own fans

> When you attack a competitor publicly, you reach two audiences: your fans, who already agree with you, and their fans, who will immediately get defensive. Your fans cheer. Their fans roll their eyes. The people actually evaluating their options – the ones who matter – are unlikely to switch to you because of a zinger of a tweet.

Reviewer 2: ```suggestion
When you attack a competitor publicly, you reach two audiences: your fans (who already agree with you) and their fans (who will immediately get defensive). The former will cheer, the latter will roll their eyes. 

The people _actually_ evaluating their options – the ones who matter – are unlikely to switch to you just because of a zinger of a tweet.
```

> You're also, the moment you go on the offensive, opening yourself up to scrutiny. We're not perfect – we've made loads of mistakes! Whatever you're accusing a competitor of – copying something, overstating a claim, making a bad call – there's a good chance you've done some version of it too, and someone will find it.

Reviewer: ```suggestion
You're also, the moment you go on the offensive, opening yourself up to scrutiny. We're not perfect and we've made loads of mistakes! Whatever you're accusing a competitor of, like copying something, overstating a claim, or making a bad call, there's a good chance you've done some version of it too, and someone will find it.
```

> You're also, the moment you go on the offensive, opening yourself up to scrutiny. We're not perfect – we've made loads of mistakes! Whatever you're accusing a competitor of – copying something, overstating a claim, making a bad call – there's a good chance you've done some version of it too, and someone will find it.

Reviewer 2: ```suggestion
You're also opening yourself up to scrutiny the moment you go on the offensive. We're not perfect; in fact, we've made loads of mistakes! Whatever you're accusing a competitor of – copying something, overstating a claim, making a bad call – there's a good chance you've done some version of it too, and someone is going to find it.
```

If you're open source, this gets even more awkward. PostHog is built on the work of many others, and we think others should be able to build on ours too. Attacking a competitor for being "inspired" by us kinda misses the point of open source.

## You're doing their marketing, with your audience

> PostHog has more users than most of our direct competitors. When we publicly name a competitor, we're lending them our audience – introducing thousands of developers to a product and telling them it's worth being upset about.

Reviewer 2: ```suggestion
PostHog has more users than most of our direct competitors. When we publicly name a competitor, we're essentially introducing thousands of developers to a product and telling them it's worth being upset about. We might as well hand over our audience to them. 
```

> So you fire off the tweet, they get more traction than they would have otherwise, their team screenshots it for the all-hands, and you've spent your energy on this instead of on your product.

Reviewer 2: ```suggestion
So you fire off the tweet, they get more traction than they would have otherwise, their team screenshots it for the all-hands, and you've spent your energy on this instead of on your product. It's a lose-lose all around.
```

## Most people didn't see the thing you're upset about

> If you're in a competitive market, you're probably monitoring it closely. Social listening, Google Alerts, the Slack channel where someone drops every mention. You are primed to notice competitor activity, and you see a lot of it.

Reviewer 2: ```suggestion
If you're in a competitive market, you're probably monitoring your audience very closely: social listening, Google Alerts, the Slack channel where someone drops every mention. You are primed to notice competitor activity, and you see a lot of it.
```

> Your users are not doing this. Most of the market didn't see the post that's been annoying you since Tuesday. Most people aren't tracking the nuances of your competitive landscape at all. [Nobody cares](https://www.youtube.com/watch?v=mmw2ub-o4gg).

Reviewer: ```suggestion
Your users are not doing this. Most people didn't see the post that's been annoying you since Tuesday. They're not tracking the nuances of your competitive landscape at all. [Nobody cares](https://www.youtube.com/watch?v=mmw2ub-o4gg).
```

> Your users are not doing this. Most of the market didn't see the post that's been annoying you since Tuesday. Most people aren't tracking the nuances of your competitive landscape at all. [Nobody cares](https://www.youtube.com/watch?v=mmw2ub-o4gg).

Reviewer 2: ```suggestion
Your users, on the other hand, are _not_ doing this on the daily. Most of them didn't see the post that's been annoying you since Tuesday. Most people aren't tracking the nuances of your competitive landscape at all. [Nobody cares](https://www.youtube.com/watch?v=mmw2ub-o4gg).
```

> The urgency of wanting to respond scales with how much it bothered you, not with how many people actually saw it. Responding publicly does more to surface the original thing than to rebut it.

Reviewer 2: ```suggestion
That sense of urgency about responding scales with how much it bothered you, not with how many people actually saw it. Don't mistake the nagging feeling for actual importance; your perception of it is bound to be inflated.
```

## What to do instead

> I'll admit - we've got this wrong before. When we threw together [isgoogleanalyticsillegal.com](https://www.isgoogleanalyticsillegal.com/), it was meant to be tongue in cheek but we still got criticized for going directly after a competitor, which was fair. 

Reviewer: ```suggestion
I'll admit, we've got this wrong before. When we threw together [isgoogleanalyticsillegal.com](https://www.isgoogleanalyticsillegal.com/), it was meant to be tongue in cheek but we still got criticized for going directly after a competitor, which was fair. 
```

> I'll admit - we've got this wrong before. When we threw together [isgoogleanalyticsillegal.com](https://www.isgoogleanalyticsillegal.com/), it was meant to be tongue in cheek but we still got criticized for going directly after a competitor, which was fair. 

Reviewer 2: ```suggestion
I'll admit – we've gotten this wrong before. When we threw together [isgoogleanalyticsillegal.com](https://www.isgoogleanalyticsillegal.com/), it was meant to be tongue-in-cheek, but we still got criticized for going directly after a competitor (which was fair). 
```

This is what we try to do instead:

> **Make fun of trends, not companies.** If something in your market is genuinely dumb, be funny and critical about that without naming anyone. The joke lands, people share it, and you haven't handed anyone ammunition.

Reviewer 2: ```suggestion
**Make fun of trends, not companies.** If something in your market is genuinely dumb, be funny and critical about that _without_ dropping any names. The joke lands better, people are more likely to share it, and you haven't handed any specific competitor any free ammunition.
```

> **Publish detailed, honest comparisons.** The instinct is to make yourself look great and them look bad - lots of ticks for you, lots of crosses for them. The better move is genuine honesty about tradeoffs – where you're stronger, where you're not, who each product is actually for. PostHog does this, and I think it builds trust that no amount of positioning copy can buy. I'll admit we haven't always got this right, so where we have made factual errors, we do our best to fix them as quickly as possible when someone notices.

Reviewer: ```suggestion
**Publish detailed, honest comparisons.** The instinct is to make yourself look great and them look bad - lots of ticks for you, lots of crosses for them. The better move is genuine honesty about tradeoffs – where you're stronger, where you're not, who each product is actually for. PostHog does this, and I think it builds trust that no amount of positioning copy can buy. I'll admit we haven't always got this right, so where we have made factual errors, we fix them as quickly as possible when they're pointed out.
```

> **Publish detailed, honest comparisons.** The instinct is to make yourself look great and them look bad - lots of ticks for you, lots of crosses for them. The better move is genuine honesty about tradeoffs – where you're stronger, where you're not, who each product is actually for. PostHog does this, and I think it builds trust that no amount of positioning copy can buy. I'll admit we haven't always got this right, so where we have made factual errors, we do our best to fix them as quickly as possible when someone notices.

Reviewer 2: ```suggestion
**Publish detailed, honest comparisons.** The instinct most people get when writing these is to make yourself look great and others look bad (i.e., lots of checkmarks for you, lots of crosses for them). It's more valuable to be genuinely honest about the tradeoffs – talk about where you're stronger, where you're not, who each product is _actually_ for. We actively do this now, and I think these build more trust than any positioning copy can buy. I'll admit we haven't always gotten these right, so whenever we have factual errors, we fix them as quickly as possible – especially when someone points them out.
```

> **Bid on competitor keywords.** If someone is searching for your competitor, you should be showing up. I know a lot of people think this doesn't work, but a) data disagrees with you, and b) sometimes we're reaching PostHog users who weren't aware we also did [thing]. You know when someone asks in a forum 'what do you think about X?' and the response is 'I actually like Y'? This is the ads equivalent of that. 

Reviewer 2: ```suggestion
**Bid on competitor keywords.** If someone is searching for your competitor, you need to show up on that list of results. A lot of people think this doesn't work, but A) data says otherwise, and B) sometimes we reach PostHog users who weren't aware we also did The Thing They Were Searching For. You know when someone asks in a forum "what do you think about X" and someone replies is "I actually like Y"? This is the ads equivalent of that. 
```

> **Correct factual errors privately if they _actually_ matter.** Contacting a competitors to say "Hey, this isn't accurate, it should be this" takes five minutes and either fixes it or motivates you when they are annoying about it. We have seen this succeed in both directions – we have happily accepted PRs to our website to set the record straight!

Reviewer: ```suggestion
**Correct factual errors privately if they _actually_ matter.** Contacting a competitor to say "Hey, this isn't accurate, it should be this" takes five minutes and either fixes it or motivates you when they are annoying about it. We have seen this succeed in both directions. We have happily accepted PRs to our website to set the record straight!
```

> **Correct factual errors privately if they _actually_ matter.** Contacting a competitors to say "Hey, this isn't accurate, it should be this" takes five minutes and either fixes it or motivates you when they are annoying about it. We have seen this succeed in both directions – we have happily accepted PRs to our website to set the record straight!

Reviewer 2: ```suggestion
**Correct people privately if the error _actually_ matters.** Contacting a competitors to say "hey, this isn't accurate, it should be this" takes five minutes and either fixes it (or motivates you further if they are annoying about fixing it). We have seen this succeed in both directions. We happily accept PRs to our website to set the record straight!
```

> But mostly: use your anger to outship them and build a better product.

Reviewer 2: ```suggestion
But the best thing to do instead: use your anger to outship them and build a better product.
```
